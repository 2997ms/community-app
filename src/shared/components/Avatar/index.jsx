/**
 * Avatar
 *
 * Description:
 *   Reusable user avatar. Depending on wether or not an image URL is
 *   provided, it will display the user's profile picture, or a default
 *   user avatar.
 *
 * Styling:
 *   Styling is applied by default (see defaultStyles object).
 *   However, the Avatar can be styled in any possible way by providing
 *   a custom style object via props.
 *   Styling implemented with react-with-styles
 *   (https://github.com/airbnb/react-with-styles)
 *
 * Usage:
 *   <Avatar url={imgUrl} customStyles="{customStylesObj}" />
 *
 * Props:
 *    - url: String, optional. Image URL.
 *    - customStyles: Object or Array of Objects, optional
 *      See https://github.com/airbnb/react-with-styles for details.
 */

import PT from 'prop-types';
import React from 'react';
import { css, withStyles } from 'utils/withStyles';
import DefaultAvatar from '../../../assets/images/ico-user-default.svg';

function Avatar(props) {
  const {
    customStyles,
    url,
  } = props;

  const defaultStyles = {
    height: '32px',
    width: '32px',
    borderRadius: '16px',
  };

  const setStyles = (styles) => {
    if (styles) {
      return styles;
    }

    return defaultStyles;
  };

  return url
    ? <img alt="Avatar" src={url} {...css(setStyles(customStyles))} />
    : <DefaultAvatar {...css(setStyles(customStyles))} />;
}

Avatar.defaultProps = {
  customStyles: null,
  url: null,
};

Avatar.propTypes = {
  customStyles: PT.oneOfType([PT.object, PT.array]),
  url: PT.string,
};

export default withStyles(() => ({}))(Avatar);
