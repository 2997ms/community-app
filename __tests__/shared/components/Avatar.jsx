import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Avatar from 'components/Avatar';

test('Matches shallow shapshot', () => {
  const baseStyles = {
    height: '100%',
    width: '100%',
  };

  const defaultModifier = {
    borderColor: 'rgba(0, 0, 0, 0.05)',
  };

  const renderer = new Renderer();
  renderer.render((
    <Avatar />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar url={'some-fake-string'} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar customStyles={baseStyles} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Avatar url={'some-fake-string'} customStyles={[baseStyles, defaultModifier]} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
