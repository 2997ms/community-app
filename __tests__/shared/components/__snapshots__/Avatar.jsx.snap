// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<Avatar
  customStyles={null}
  styles={Object {}}
  theme={Object {}}
  url={null}
/>
`;

exports[`Matches shallow shapshot 2`] = `
<Avatar
  customStyles={null}
  styles={Object {}}
  theme={Object {}}
  url="some-fake-string"
/>
`;

exports[`Matches shallow shapshot 3`] = `
<Avatar
  customStyles={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  styles={Object {}}
  theme={Object {}}
  url={null}
/>
`;

exports[`Matches shallow shapshot 4`] = `
<Avatar
  customStyles={
    Array [
      Object {
        "height": "100%",
        "width": "100%",
      },
      Object {
        "borderColor": "rgba(0, 0, 0, 0.05)",
      },
    ]
  }
  styles={Object {}}
  theme={Object {}}
  url="some-fake-string"
/>
`;
